base/src/com/google/idea/blaze/base/command/info/BlazeInfoProvider.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2023 The Bazel Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.idea.blaze.base.command.info;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.idea.blaze.base.async.executor.BlazeExecutor;
+import com.google.idea.blaze.base.bazel.BuildSystem;
+import com.google.idea.blaze.base.scope.BlazeContext;
+import com.google.idea.blaze.base.settings.Blaze;
+import com.google.idea.blaze.base.settings.BlazeImportSettings;
+import com.google.idea.blaze.base.settings.BlazeImportSettingsManager;
+import com.google.idea.blaze.base.sync.SyncListener;
+import com.google.idea.blaze.base.sync.SyncMode;
+import com.google.idea.blaze.base.sync.SyncResult;
+import com.google.idea.common.experiments.BoolExperiment;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.project.Project;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * <p> Providing a {@link com.google.idea.blaze.base.command.info.BlazeInfo}, the result will be cached and invalidated on any sync failure. </p>
+ *
+ * @see BlazeInfoRunner
+ */
+public class BlazeInfoProvider {
+    private static final Logger logger = Logger.getInstance(BlazeInfoProvider.class);
+    private static final BoolExperiment enabled =
+            new BoolExperiment("blaze.info.provider.enabled", false);
+
+    private final Project project;
+    private volatile BlazeInfo blazeInfo;
+
+    public BlazeInfoProvider(Project project) {
+        this.project = project;
+    }
+
+    public static BlazeInfoProvider getInstance(Project project) {
+        return project.getService(BlazeInfoProvider.class);
+    }
+
+    public static Boolean isEnabled() {
+        return enabled.getValue();
+    }
+
+    public ListenableFuture<String> getBlazeInfo(
```

Comment:
Should this be public? Taking a key seems like an implementation detail.

Reply:
Maybe we can remove this method altogether, the idea was to offer same functionality as [BlazeInfoRunner](https://github.com/bazelbuild/intellij/blob/17b0bb8147bd11642c11cbce325ac9f301bca402/base/src/com/google/idea/blaze/base/command/info/BlazeInfoRunner.java#L27), but currently this method is not being used.

Reply:
Removed.